src/test/java/org/opensearch/ad/HistoricalAnalysisRestTestCase.java

```diff
@@ -62,7 +63,7 @@ public ToXContentObject[] getHistoricalAnomalyDetector(String detectorId, boolea
         return getAnomalyDetector(detectorId, header, false, returnTask, client);
     }
 
-    public ADTaskProfile getADTaskProfile(String detectorId) throws IOException {
+    public ADTaskProfile getADTaskProfile(String detectorId) throws IOException, ParseException {
```

Comment:
What changed where this can throw a `ParseException` now?

Reply:
the toString() method in `EntityUtils.toString(profileResponse.getEntity());`

Reply:
Was this due to a dependency change? Can you link the method that has changed? I don't see anything explicitly changed here

Reply:
```
    /**
     * Reads the contents of an entity and return it as a String.
     * The content is converted using the character set from the entity (if any),
     * failing that, "ISO-8859-1" is used.
     *
     * @param entity the entity to convert to a string; must not be null
     * @return String containing the content.
     * @throws ParseException if header elements cannot be parsed
     * @throws IllegalArgumentException if entity is null or if content length &gt; Integer.MAX_VALUE
     * @throws IOException if an error occurs reading the input stream
     * @throws java.nio.charset.UnsupportedCharsetException Thrown when the named charset is not available in
     * this instance of the Java virtual machine
     */
    public static String toString(final HttpEntity entity) throws IOException, ParseException {
        return toString(entity, DEFAULT_ENTITY_RETURN_MAX_LENGTH);
    }
```

Reply:
Is this from one of the apache dependency changes? Sorry for the confusion just wondering where this changed exactly.

Reply:
yes, it's from the apache dependency change. You can find the import of this exception is `org.apache.hc.core5.http.ParseException`

Reply:
I was just wondering where that method was coming from - I now see it's from `EntityUtils` which is a dependency